### ISSUE TITLE
Add some check on EmitterSubscription's listener

### ISSUE THREAD

**Issue by em-yuanbo** [facebook/emitter#25]
https://github.com/facebook/emitter/blob/master/src/EmitterSubscription.js#L32
there is no validation on the param `listener`
>```javascript
  constructor(subscriber: EventSubscriptionVendor, listener, context: ?Object) {
    super(subscriber);
    this.listener = listener;
    this.context = context;
  }
```

and no `appliable` check when apply the listener. 
https://github.com/facebook/emitter/blob/master/src/BaseEventEmitter.js#L182
>```javascript
  __emitToSubscription(subscription, eventType) {
    var args = Array.prototype.slice.call(arguments, 2);
    subscription.listener.apply(subscription.context, args);
  }
```

can we add some checks or make some noises when create a new EmitterSubscription?

**Comment by kyldvs** [comment]
The file has typechecks which will complain if you pass anything other than a function to it. We should also probably add flow types too, but I do not believe we should modify runtime behavior to verify it's a function.

**Comment by em-yuanbo** [comment]
The typechecks or the flow types only make sense when you compile this source code. But most time the libs  we used  are installed from npm  and compiled, and all the types or checks are missed. by then the condition is different.

**Comment by kyldvs** [comment]
Yeah that will be true of typechecks probably, but for flow we can provide external library definitions. Doesn't matter if the module is transpiled or not
